camayoc/tests/qpc/ui/test_sources.py

```diff
@@ -11,17 +11,44 @@
 """
 import pytest
 
-from .views import DashboardView
+from camayoc import utils
 
+from .utils import create_source, delete_source
 
-@pytest.mark.skip(reason='Test stub')
-def test_create_source(browser):
-    """Create a source through the UI.
+SOURCE_DATA = {
+    'Network': [
+        '127.0.0.1',
+        '127.0.0.1, 127.0.0.2',
+        '192.168.1.[1:100]',
+        '192.168.0.0/24',
+        'example.sonar.com'],
+    'Satellite': [
+        '127.0.0.1',
+        'examplesatellite.sonar.com'],
+    'VCenter': [
+        '127.0.0.1',
+        'examplevcenter.sonar.com'],
+    }
+
+
+@pytest.mark.parametrize('source_type, ', SOURCE_DATA.keys())
+def test_create_delete_source(browser, qpc_login, credentials, source_type):
+    """Create and then delete a source through the UI.
 
     :id: b1f64fd6-0421-4650-aa6d-149cb3099012
     :description: Creates a source in the UI.
-    :steps: TODO
-    :expectedresults: TODO
+    :steps:
+        1) Go to the sources page and open the sources modal.
+        2) Fill in the required information and create a new source.
+        3) Remove the newly created source.
+    :expectedresults: A new source is created with the provided information,
+        then it is deleted.
     """
-    dash = DashboardView(browser)
-    dash.nav.select('Sources')
+    source_name = utils.uuid4()
+    credential_name = credentials[source_type]
+    for addresses in SOURCE_DATA[source_type]:
+        create_source(
+                browser, credential_name, source_type, source_name, addresses)
+        #  Deletion implicitly asserts new sources exists in the UI.
+        delete_source(browser, source_name)
```

Comment:
This doesn't get a chance to test that deletion works when there is more than one source, since it always delete it immediately. If you created all the sources, _then_ deleted them all afterwards, the test would cover a bit more cases.

Reply:
Good call, I'll make that adjustment.

Reply:
@ironfroggy I remember why this wasn't done...the UI doesn't support multi-deletion :smile: